src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -375,124 +395,246 @@ emit_sri_vector128 (TransformData *td, MonoMethod *cmethod, MonoMethodSignature
 	gint16 simd_opcode = -1;
 	gint16 simd_intrins = -1;
 
-	vector_klass = mono_class_from_mono_type_internal (csignature->ret);
+	MonoTypeEnum ret_atype;
```

Comment:
In here, we consistently query the relevant simd information of the return type and if it exists the simd information of the  first parameter (this is enough to correctly handle all the cases that currently exist).

There's actually quite a bit of logic in this function that *could* be moved down into `emit_common_simd_operations`, as APIs like `AndNot` exist for `Vector128<T>` and `Vector<T>`, they may also exist for types like `Vector4` in the future. I opted to not move that down in this PR, to try and keep the total churn under control.

But, I did add some basic validation that the encountered signatures are roughly as expected to help ensure we don't hit issues in the future as new overloads are introduced or the general SIMD support in Mono is expanded.